### PR TITLE
Added in grid view in filebrowser

### PR DESCRIPTION
**Pull request by bertmeeuws**
@bertmeeuws
Refactoring code, and styling the contact add for visual consistency #… 

Fixed alignment

New changes since you last viewed
Adding hovers, visual enhancements and UX #123 #122 #124 #128

Added grid view for shared with me

added empty state in grid view

Grid in local filebrowser WOIP

Added a shared with me placeholder

Fixing state error

- Add grid view with empty state for shared with me & local
- Added warnings for example if a user uses a '/' in a folder name.
- Restyled the modals for visual consistency
- Added some hovers on icons

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -4,23 +4,19 @@
         <meta charset="utf-8" />
         <meta http-equiv="X-UA-Compatible" content="IE=edge" />
         <meta name="viewport" content="width=device-width,initial-scale=1.0" />
-        <link rel="icon" href="/logo_no_stroke.png"/>
+        <link rel="icon" href="/logo_no_stroke.png" />
```
Comment (JonasWijne) [inline index.html:7]: weird

File: public/config/version.ts [hunk 2]
```diff
@@ -1 +1 @@
-export default "{{version}}";
\ No newline at end of file
+export default '{{version}}';
```
Comment (JonasWijne) [inline public/config/version.ts:1]:
```suggestion
export default "{{version}}";
```

Comment (JonasWijne) [reply public/config/version.ts:1]: not necessary

File: .prettierrc [hunk 3]
```diff
@@ -1,10 +1,10 @@
 {
-  "trailingComma": "es5",
-  "tabWidth": 4,
-  "semi": true,
-  "singleQuote": true,
-  "arrowParens": "avoid",
-  "proseWrap": "preserve",
-  "printWidth": 120,
-  "vueIndentScriptAndStyle": true
+    "trailingComma": "es5",
```
Comment (JonasWijne) [inline .prettierrc:2]: whitespace => pls change

File: README.md [hunk 4]
```diff
@@ -4,6 +4,6 @@ Experience as build on top of the digital twin (see twin_server repo).
 
 This experience has following functionality
 
-- chat interface
-- file manager interface
-- launcher video conferencing
+-   chat interface
```
Comment (JonasWijne) [inline README.md:7]: whitespace

File: index.html (hunk 1 above)
Comment (JonasWijne) [inline index.html:7]: ?

File: index.html [hunk 5]
```diff
@@ -4,23 +4,19 @@
...
         <title>Uhuru</title>
-        <link
-            href="https://use.fontawesome.com/releases/v5.1.1/css/all.css"
-            rel="stylesheet"
-        />
+        <link href="https://use.fontawesome.com/releases/v5.1.1/css/all.css" rel="stylesheet" />
```
Comment (JonasWijne) [inline index.html:9]: ?

File: index.html [hunk 6]
```diff
@@ -4,23 +4,19 @@
...
         <meta http-equiv="ScreenOrientation" content="autoRotate:disabled" />
         <meta />
         <script type="module" src="/config/config.js"></script>
-        <script type="module" src='/src/main.ts'></script>
+        <script type="module" src="/src/main.ts"></script>
     </head>
     <body class="h-full">
         <noscript>
             <strong
-                >We're sorry but <%= htmlWebpackPlugin.options.title %> doesn't
```
Comment (JonasWijne) [inline index.html:21]: ?

File: index.html [hunk 7]
```diff
@@ -4,23 +4,19 @@
...
-                work properly without JavaScript enabled. Please enable it to
-                continue.</strong
+                >We're sorry but <%= htmlWebpackPlugin.options.title %> doesn't work properly without JavaScript
+                enabled. Please enable it to continue.</strong
```
Comment (JonasWijne) [inline index.html:19]: ?

File: public/config/config.js [hunk 8]
```diff
@@ -1,5 +1,5 @@
-import config from '/config/config.json' assert { type: "json" };
+import config from '/config/config.json' assert { type: 'json' };
```
Comment (JonasWijne) [inline public/config/config.js:1]: ?

File: public/config/config.js [hunk 9]
```diff
@@ -1,5 +1,5 @@
...
 
-config.baseUrl =  `${window.location.origin}/`
+config.baseUrl = `${window.location.origin}/`;
```
Comment (JonasWijne) [inline public/config/config.js:3]: ?

File: public/config/config.js [hunk 10]
```diff
@@ -1,5 +1,5 @@
...
 
-window.config = config
\ No newline at end of file
+window.config = config;
```
Comment (JonasWijne) [inline public/config/config.js:5]: ?

File: public/config/config.json [hunk 11]
```diff
@@ -3,5 +3,4 @@
     "documentServerLocation": "https://documentserver.digitaltwin-test.jimbertesting.be/",
     "giphyApiKey": "uk3XRSO0vYrPDEQKDPZJ2wGz33qzIxST",
     "beta": true
-
```
Comment (JonasWijne) [inline public/config/config.json:6]: unnecessary

File: src/assets/index.css [hunk 12]
```diff
@@ -22,7 +22,7 @@
     label {
         white-space: nowrap;
     }
-    th{
+    th {
```
Comment (JonasWijne) [inline src/assets/index.css:25]: ?

File: src/components/AudioPlayer.vue [hunk 13]
```diff
@@ -1,39 +1,54 @@
 <template>
-    <div class='flex items-center p-2 pb-6'>
-        <audio ref='audiofile' :src='src' class='hidden' preload='auto'></audio>
+    <div class="flex items-center p-2 pb-6">
```
Comment (JonasWijne) [inline src/components/AudioPlayer.vue:2]: what is standard: make ticket to check all files follow up with project lead

File: src/components/AudioPlayer.vue [hunk 14]
```diff
@@ -1,39 +1,54 @@
...
+        <audio ref="audiofile" :src="src" class="hidden" preload="auto"></audio>
         <button
-            v-if='isPlaying'
-            class='ml-0.5 bg-white p-1.5 rounded-full text-gray-400 hover:text-accent-400 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-accent-500'
-            type='button'
-            @click='isPlaying = false'
+            v-if="isPlaying"
+            class="
```
Comment (JonasWijne) [inline src/components/AudioPlayer.vue:6]: what is standard: make ticker to check all files

File: src/components/AudioPlayer.vue [hunk 15]
```diff
@@ -1,39 +1,54 @@
...
+                text-gray-400
+                hover:text-accent-400
+                focus:outline-none focus:ring-2 focus:ring-inset focus:ring-accent-500
+            "
+            type="button"
+            @click="isPlaying = false"
         >
-            <PauseIcon aria-hidden='true' class='h-5 w-5' />
-            <span class='sr-only'>Pause</span>
+            <PauseIcon aria-hidden="true" class="h-5 w-5" />
```
Comment (JonasWijne) [inline src/components/AudioPlayer.vue:18]: ?

File: src/components/AudioPlayer.vue [hunk 16]
```diff
@@ -1,39 +1,54 @@
...
+            <span class="sr-only">Pause</span>
         </button>
         <button
             v-else
-            class='ml-0.5 bg-white p-1.5 rounded-full text-gray-400 hover:text-accent-400 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-accent-500'
-            type='button'
-            @click='isPlaying = true'
+            class="
```
Comment (JonasWijne) [inline src/components/AudioPlayer.vue:23]: ?

File: src/components/AudioPlayer.vue [hunk 17]
```diff
@@ -1,39 +1,54 @@
...
+                ml-0.5
+                bg-white
+                p-1.5
+                rounded-full
+                text-gray-400
+                hover:text-accent-400
+                focus:outline-none focus:ring-2 focus:ring-inset focus:ring-accent-500
+            "
+            type="button"
+            @click="isPlaying = true"
```
Comment (JonasWijne) [inline src/components/AudioPlayer.vue:33]: ?

File: src/components/Button.vue [hunk 18]
```diff
@@ -1,5 +1,7 @@
 <template>
-    <button class="flex items-center rounded-xl m-2 bg-accent-300 text-white btn disabled:bg-gray-300 bg-cursor-[unset]">
+    <button
```
[the remark this answers is not in the document]
Comment (JonasWijne) [reply src/components/Button.vue:2]: why did you add this file please make sure you do git add with `git add -p`

File: src/components/ChatCard.vue [hunk 19]
```diff
@@ -5,39 +5,49 @@
...
+            hover:bg-gray-200
+            transition
+            duration-100
+        "
+        tabindex="0"
+        @click="$emit('selectChat')"
+        @keyup.enter="$emit('selectChat')"
     >
-        <div class='place-items-center relative'>
+        <div class="place-items-center relative">
```
Comment (JonasWijne) [inline src/components/ChatCard.vue:23]: ?

**Review by JonasWijne — COMMENTED** [review]
this merge conflict is bad please checkl first before comming back to me

**Comment by JonasWijne** [comment]
commit messages not specific enough

**Comment by bertmeeuws** [comment]
- Adding hovers on sidebars and modals
- Added cleaner looking exit button in modal
- Clicking outside of the modal will exit it
- ESCAPE key in modal will also exit it
- Using a "/" in a folder name will give real time feedback
- Added an error when a has no name
- Added a max. char length of 50 in a folder name
- When a file is now dragged into the fileuploader it doesn't reset, it takes the exisiting files in the array and adds the new one
- Put in error saying you need atleast one file when trying to upload if none are chosen
- Added a small fix for when /quantum is loaded but the directory isn't set to '/'
- Added grid mode for local filebrowser and shared with me folder
- Refactored the navigation in ContactAdd.vue to reading it out from an array and not hardcoded.
- You can now unselect folders/files by clicking the gray area

image

image

image

image

**Review by JonasWijne — COMMENTED** [review]
too much unnecessary changes

File: src/components/ChatCard.vue [hunk 20]
```diff
@@ -5,39 +5,49 @@
             'bg-gray-200 border border-gray-50': router.currentRoute?.value.path.includes(chat.chatId),
             'opacity-50': blocked,
         }"
-        class='chatcard relative text-sm flex flex-row focus:outline-none focus:ring-4 ring-accent-500 ring-offset-2'
-        tabindex='0'
-        @click='$emit("selectChat")'
-        @keyup.enter='$emit("selectChat")'
+        class="
```
[the remark this answers is not in the document]
Comment (bertmeeuws) [reply src/components/ChatCard.vue:8]: Added a hover & duration

File: src/components/ChatList.vue [hunk 21]
```diff
@@ -22,7 +22,18 @@
                 <div class="flex-1 collapsed-bar:mb-2 flex flex-row items-center">
                     <button
                         @click="showAddUserDialog = true"
-                        class="bg-primary rounded-full text-white w-8 h-8 mx-2 collapsed-bar:w-10 collapsed-bar:h-10"
+                        class="
+                            bg-primary
+                            hover:bg-accent-800
+                            transition
+                            duration:300
```
Comment (bertmeeuws) [inline src/components/ChatList.vue:29]: Added a hover & duration

File: src/components/ContactAdd.vue [hunk 22]
```diff
@@ -2,23 +2,20 @@
     <div class="place-items-start">
         <div class="grid grid-cols-2">
             <a
-                :class="{ active: isActive('user') }"
-                class="nav-link grid-cols-6 text-center py-2"
+                v-for="item in navigation"
```
Comment (bertmeeuws) [inline src/components/ContactAdd.vue:5]: Navigation is now read out from an array

File: src/components/ContactAdd.vue [hunk 23]
```diff
@@ -170,12 +167,27 @@
...
+                        py-2
+                        px-4
+                        ml-2
+                        text-white
+                        rounded-md
+                        justify-self-end
+                        bg-primary
+                        hover:bg-accent-700
+                        transition
+                        duration-300
```
Comment (bertmeeuws) [inline src/components/ContactAdd.vue:186]: Added a hover & duration

File: src/components/Dialog.vue [hunk 24]
```diff
@@ -22,9 +24,25 @@
...
                 <div class="flex justify-between">
                     <slot name="title" />
-                    <button @click="$emit('update-model-value', false)">
-                        <i class="fas fa-times"></i>
-                    </button>
+                    <div
+                        class="
+                            w-6
+                            h-6
+                            bg-transparent
```
Comment (bertmeeuws) [inline src/components/Dialog.vue:31]: - Added a circle around the exit button in a modal
- Added a hover & duration

File: src/components/Dialog.vue [hunk 25]
```diff
@@ -37,7 +55,18 @@
...
+                            py-2
+                            px-4
+                            ml-2
+                            text-white
+                            rounded-md
+                            justify-self-end
+                            bg-primary
+                            hover:bg-accent-700
+                            transition
+                            duration-300
```
Comment (bertmeeuws) [inline src/components/Dialog.vue:68]: Added a hover & duration

File: src/components/Sidebar.vue [hunk 26]
```diff
@@ -15,7 +15,7 @@
                 v-for="app in apps"
                 :key="app.name"
                 :class="{
-                    'cursor-pointer': app.enabled,
+                    'cursor-pointer transition duration-300 hover:bg-primarylight rounded-xl': app.enabled,
```
Comment (bertmeeuws) [inline src/components/Sidebar.vue:18]: Added a hover & duration

File: src/components/UserTable.vue [hunk 27]
```diff
@@ -67,7 +67,17 @@
...
+                                            class="
+                                                text-white
+                                                py-2
+                                                px-4
+                                                rounded-md
+                                                justify-self-end
+                                                bg-primary
+                                                hover:bg-accent-700
+                                                transition
+                                                duration-300
```
Comment (bertmeeuws) [inline src/components/UserTable.vue:79]: Added a hover & duration

File: src/components/UserTableGroup.vue [hunk 28]
```diff
@@ -64,7 +64,17 @@
...
-                                                class="text-white py-2 px-4 rounded-md justify-self-end bg-primary"
+                                                class="
+                                                    text-white
+                                                    py-2
+                                                    px-4
+                                                    rounded-md
+                                                    justify-self-end
+                                                    bg-primary
+                                                    hover:bg-accent-700
+                                                    transition
```
Comment (bertmeeuws) [inline src/components/UserTableGroup.vue:75]: Added a hover & duration

File: src/components/fileBrowser/FileTable.vue [hunk 29]
```diff
@@ -1,17 +1,21 @@
...
         <div class="overflow-x-auto">
-            <div class="py-2 align-middle inline-block min-w-full">
-                <div class="shadow overflow-hidden border-b border-gray-200 sm:rounded-lg">
-                    <FileDropArea class="h-full" @send-file="uploadFiles">
+            <div class="py-2 px-4 align-middle inline-block min-w-full">
+                <div class="flex justify-end mb-2">
+                    <ViewSelect />
+                </div>
+                <div class="overflow-hidden border-b border-gray-200 sm:rounded-lg">
+                    <FileDropArea @click.stop class="h-full" @send-file="uploadFiles">
```
Comment (bertmeeuws) [inline src/components/fileBrowser/FileTable.vue:9]: Added a @click.stop to propagate the "deselect all files by clicking the gray background" when clicking around in the filebrowser itself

File: src/components/fileBrowser/FileTable.vue [hunk 30]
```diff
@@ -182,6 +185,130 @@
                                 </tr>
                             </tbody>
                         </table>
+                        <ul
```
Comment (bertmeeuws) [inline src/components/fileBrowser/FileTable.vue:188]: This is the new grid view

File: src/components/fileBrowser/FileTable.vue [hunk 31]
```diff
@@ -308,6 +437,10 @@
                 isDraggingFiles.value = false;
             };
 
+            const goBack = () => {
```
Comment (bertmeeuws) [inline src/components/fileBrowser/FileTable.vue:440]: When in grid view and a folder is empty you get the option to go back one page

File: src/components/fileBrowser/MainActionButtons.vue [hunk 32]
```diff
@@ -1,17 +1,26 @@
...
+        <button
+            @click="showCreateFolderDialog = true"
+            class="text-white py-2 px-4 mr-2 rounded-md bg-primary hover:bg-accent-800 transition duration:300"
+        >
             <i class="fas fa-plus"></i> New Folder
         </button>
-        <button @click="showCreateFileDialog = true" class="text-white py-2 px-4 mr-2 rounded-md bg-primary">
-            <i class="fas fa-plus"></i> Upload Files
+        <button
+            @click="showCreateFileDialog = true"
```
Comment (bertmeeuws) [inline src/components/fileBrowser/MainActionButtons.vue:10]: Added a hover & duration

File: src/components/fileBrowser/MainActionButtons.vue [hunk 33]
```diff
@@ -1,17 +1,26 @@
 <template>
     <div class="mx-2" :class="{ hidden: sharedDir }">
-        <button @click="showCreateFolderDialog = true" class="text-white py-2 px-4 mr-2 rounded-md bg-primary">
+        <button
+            @click="showCreateFolderDialog = true"
```
Comment (bertmeeuws) [inline src/components/fileBrowser/MainActionButtons.vue:4]: Added a hover & duration

File: src/components/fileBrowser/MainActionButtons.vue [hunk 34]
```diff
@@ -95,18 +107,34 @@
...
+            const createFolderErrors = ref<string[]>([]);
+            const manualContactAdd = ref<string>('');
+            const fileUploadErrors = ref<string[]>([]);
+
+            watch(manualContactAdd, () => {
+                createFolderErrors.value = [];
+                if (manualContactAdd.value.includes('/')) {
+                    createFolderErrors.value.push("'/' are not allowed in folder names.");
+                }
+                if (manualContactAdd.value.length >= 50) {
```
Comment (bertmeeuws) [inline src/components/fileBrowser/MainActionButtons.vue:119]: - Added max. char length in folders and it checks for '/'

**Comment by bertmeeuws** [comment]
@JonasWijne Please sir, review my code

File: src/store/fileBrowserStore.ts [hunk 35]
```diff
@@ -64,6 +69,7 @@ export const selectedAction = ref<Action>(Action.COPY);
 export const sharedDir = ref(false);
 export const sharedContent = ref<SharedFileInterface[]>([]);
 export const allSharedContent = ref<SharedFileInterface[]>([]);
+export const fileBrowserTypeView = ref<string>('LIST');
```
Comment (JonasWijne) [inline src/store/fileBrowserStore.ts:72]: should be enum
```suggestion
export const fileBrowserTypeView = ref<View>('LIST');
```

File: src/store/fileBrowserStore.ts [hunk 36]
```diff
@@ -38,6 +38,11 @@ export enum Action {
     COPY,
 }
 
+export enum View {
+    LIST,
```
Comment (JonasWijne) [inline src/store/fileBrowserStore.ts:42]:
```suggestion
    LIST = "LIST",
```

**Review by JonasWijne — CHANGES_REQUESTED** [review]

**Pull request merged**